code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -338,6 +342,10 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	weldermes = "<span class='notice'>USER insults NAME by lighting it with FLAME.</span>"
 	ignitermes = "<span class='notice'>USER fiddles with FLAME, and manages to light their NAME with the power of science.</span>"
 
+	New()

```

Comment:
Relative pathing pls no.

Reply:
Would it be better as 
`/obj/item/clothing/mask/smokable/cigarette/cigar/New()
	..()
	reagents.add_reagent(/datum/reagent/nicotine, 5))`
or am I misunderstanding what you're saying?

Reply:
Yes, the Github's wiki tab has examples of absolute and relative pathing, under the Standards section.

Reply:
Also, that ..() calls the New() proc from the item above it, pathwise, which is in this case /obj/item/clothing/mask/smokable/cigarette/New(), so if you've added nicotine there, it will also be called in this proc... Which may or may not give you the intended amount of nicotine.

Reply:
I'll be honest, I didn't understand what the non-absolute pathing thing meant the first time I read it there, and I still don't understand what the hell that is or how it relates. I might be a moron.

Reply:
Relative pathing is when you define a proc [in this case New()] in the object's code, without specifying in the line which object the proc is supposed to function for.

If you have something with the path: /obj/item/thing, 

/obj/item/thing/New()
is absolute pathing. You can tell exactly which item will call this proc, because it's there on the line.

-> New()
is relative pathing. This is less than ideal, because it doesn't directly point you to the item it's supposed to work with. This isn't a huge problem with simple items that might only have one or two procs, but if it gets more complicated it really starts to become an issue. Also it's very easy to lose procs through poor indentation. Overall, we like to avoid using relative pathing.

Reply:
That explains it a lot better, thank you very much.